src/node/kernel_notifications.cpp

```diff
@@ -99,6 +100,14 @@ void KernelNotifications::fatalError(const bilingual_str& message)
                     m_exit_status, message, &m_warnings);
 }
 
+std::optional<uint256> KernelNotifications::TipBlock()
+{
+    AssertLockHeld(m_tip_block_mutex);
+    Assume(!m_tip_block || m_tip_block.value() != uint256{});
```

Comment:
nit:
Using `uint256::ZERO` seems clearer to me, but I don't feel too strongly about it.

Reply:
I droppend this check, only keeping the setter.